src/common/guac_surface.c

```diff
@@ -437,8 +437,8 @@ static void __guac_common_surface_put(unsigned char* src_buffer, int src_stride,
 
     int x, y;
 
-    int min_x = rect->width - 1;
```

Comment:
Ah - right. Well, looking back at what I originally said, I was *way* off base:

> If `min_x` and `min_y` are truly to be the minimum possible values of X and Y within the surface, as their name and usage imply, then they are correctly calculated as `rect->width - 1` and `rect->height - 1` respectively.

... which is completely wrong and backwards. It's mostly true for `max_x` and `max_y`, definitely not `min_x` and `min_y`.

The original intent of those lines was, no doubt, to choose values outside the possible range of X and Y for the sake of calculating a dirty rect, which is exactly what your change does and my original code didn't.

Looks good to me, and thanks for the clarification.